### ISSUE TITLE
D&A loss & acc fixed and dead,why?

### ISSUE THREAD

**Issue by taichu012** [roatienza/Deep-Learning-Experiments#2]
I just run this python program of DCGAN with keras, it seems work and after 20 mins, all outputs and numbers are fixed and NEVER changed again, it seems dead, but WHY? output is copied below:

48: [D loss: 7.971192, acc: 0.500000]  [A loss: 0.000000, acc: 1.000000]
49: [D loss: 7.971192, acc: 0.500000]  [A loss: 0.000000, acc: 1.000000]
50: [D loss: 7.971192, acc: 0.500000]  [A loss: 0.000000, acc: 1.000000]
51: [D loss: 7.971192, acc: 0.500000]  [A loss: 0.000000, acc: 1.000000]
52: [D loss: 7.971192, acc: 0.500000]  [A loss: 0.000000, acc: 1.000000]
53: [D loss: 7.971192, acc: 0.500000]  [A loss: 0.000000, acc: 1.000000]
54: [D loss: 7.971192, acc: 0.500000]  [A loss: 0.000000, acc: 1.000000]
55: [D loss: 7.971192, acc: 0.500000]  [A loss: 0.000000, acc: 1.000000]
56: [D loss: 7.971192, acc: 0.500000]  [A loss: 0.000000, acc: 1.000000]
57: [D loss: 7.971192, acc: 0.500000]  [A loss: 0.000000, acc: 1.000000]
58: [D loss: 7.971192, acc: 0.500000]  [A loss: 0.000000, acc: 1.000000]
59: [D loss: 7.971192, acc: 0.500000]  [A loss: 0.000000, acc: 1.000000]
60: [D loss: 7.971192, acc: 0.500000]  [A loss: 0.000000, acc: 1.000000]
61: [D loss: 7.971192, acc: 0.500000]  [A loss: 0.000000, acc: 1.000000]
62: [D loss: 7.971192, acc: 0.500000]  [A loss: 0.000000, acc: 1.000000]
63: [D loss: 7.971192, acc: 0.500000]  [A loss: 0.000000, acc: 1.000000]
64: [D loss: 7.971192, acc: 0.500000]  [A loss: 0.000000, acc: 1.000000]
65: [D loss: 7.971192, acc: 0.500000]  [A loss: 0.000000, acc: 1.000000]
66: [D loss: 7.971192, acc: 0.500000]  [A loss: 0.000000, acc: 1.000000]
67: [D loss: 7.971192, acc: 0.500000]  [A loss: 0.000000, acc: 1.000000]

**Comment by roatienza** [comment]
I adjusted the learning rate of RMSProp. Pls try again. The output near the end of 10k iteration should appear something like this:

9985: [D loss: 0.697829, acc: 0.535156]  [A loss: 0.771172, acc: 0.343750]
9986: [D loss: 0.699841, acc: 0.513672]  [A loss: 0.792130, acc: 0.296875]
9987: [D loss: 0.711574, acc: 0.507812]  [A loss: 0.842314, acc: 0.199219]
9988: [D loss: 0.697459, acc: 0.521484]  [A loss: 0.801781, acc: 0.253906]
9989: [D loss: 0.691388, acc: 0.521484]  [A loss: 0.810089, acc: 0.269531]
9990: [D loss: 0.713812, acc: 0.490234]  [A loss: 0.838145, acc: 0.210938]
9991: [D loss: 0.703769, acc: 0.478516]  [A loss: 0.783158, acc: 0.335938]
9992: [D loss: 0.694213, acc: 0.531250]  [A loss: 0.812030, acc: 0.246094]
9993: [D loss: 0.707447, acc: 0.503906]  [A loss: 0.876013, acc: 0.156250]
9994: [D loss: 0.689719, acc: 0.556641]  [A loss: 0.780946, acc: 0.281250]
9995: [D loss: 0.707479, acc: 0.521484]  [A loss: 0.905610, acc: 0.101562]
9996: [D loss: 0.692887, acc: 0.529297]  [A loss: 0.709554, acc: 0.511719]
9997: [D loss: 0.702180, acc: 0.523438]  [A loss: 0.932314, acc: 0.101562]
9998: [D loss: 0.698255, acc: 0.519531]  [A loss: 0.685984, acc: 0.542969]
9999: [D loss: 0.711334, acc: 0.515625]  [A loss: 0.925216, acc: 0.085938]

**Comment by speedyray** [comment]
How long does it take to run 10K iterations on a 2.7 GHZ intel Core i5 processor Mac book?

I was at 150 iterations after so many hours? Is there a problem with code, system or I need a GPU?

**Comment by nihitpatel** [comment]
Mine system is having 16 gb of RAM and core i7 processor. For 50 iterations
it took nearly 25 to 30 minutes. So for 10k iterations it could take even
days. And yeah it is highly dependent on the kind of hardware you have. I
would tell you to rent a cloud computer for doing the job.
